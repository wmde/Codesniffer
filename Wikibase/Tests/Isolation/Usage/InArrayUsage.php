<?php

// Should use array_key_exists
in_array( $key, array_keys( $array ) );
in_array( $key, array_flip( $array ) );
in_array( $key, array_keys );

// Nothing wrong with these
in_array( $key, $array );
array_keys( $array );
array_key_exists( $key, $array );
in_array( $key ) && array_keys( $array );
in_array( array_keys( $key ), $array );

// Nested parenthesis
if ( !in_array( $key, array_keys( $array ) ) ) {
}
