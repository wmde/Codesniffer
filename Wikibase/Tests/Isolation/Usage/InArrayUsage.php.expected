  4 | ERROR | Found slow in_array( array_keys ), should be array_key_exists or isset
  5 | ERROR | Found slow in_array( array_flip ), should be array_key_exists or isset
  6 | ERROR | Found slow in_array( array_keys ), should be array_key_exists or isset
 16 | ERROR | Found slow in_array( array_keys ), should be array_key_exists or isset
